### PR TITLE
Document connect and sign

### LINKED ISSUE #995: Document SDK connect and sign
Document the JS SDK connect and sign functionality. Probably add this to the "sign data" category.

### PR DESCRIPTION
**Pull request by alexandratran**
Add how-to page for using the `connectAndSign` method of the JS and iOS SDK. Fixes #995.

Preview: https://docs.metamask.io/995-sdk-connect-and-sign/wallet/how-to/sign-data/connect-and-sign/

### CODE REVIEW

**Review by andreahaku — APPROVED** [review]
LGTM

File: wallet/how-to/connect/set-up-sdk/javascript/react/index.md [hunk 1]
```diff
@@ -154,5 +154,8 @@ const connect = async () => {
 };
 ```
 
+You can also [use the `connectAndSign` method](../../../../sign-data/connect-and-sign.md) to
+initiate a connection and request to sign data in a single interaction.
```
Comment (bgravenorst) [inline wallet/how-to/connect/set-up-sdk/javascript/react/index.md:158]:
```suggestion
You can also [use the `connectAndSign` method](../../../../sign-data/connect-and-sign.md) to
connect to MetaMask and sign data in a single interaction.
```

File: wallet/how-to/sign-data/connect-and-sign.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+---
+description: Use MetaMask SDK to connect and sign in a single interaction.
+sidebar_position: 1
+---
+
+# Connect and sign
+
+You can connect to MetaMask and sign data in a single interaction from your JavaScript or iOS dapp
+with [MetaMask SDK](../../concepts/sdk/index.md) installed.
```
Comment (bgravenorst) [inline wallet/how-to/sign-data/connect-and-sign.md:9]:
```suggestion
You can connect to MetaMask and sign data in a single interaction from your JavaScript or iOS dapp
using the [MetaMask SDK](../../concepts/sdk/index.md).
```

File: wallet/how-to/sign-data/connect-and-sign.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+The SDK provides a `connectAndSign` method, which is a streamlined approach for dapps to interact
+with MetaMask.
```
Comment (bgravenorst) [inline wallet/how-to/sign-data/connect-and-sign.md:12]:
```suggestion
The SDKs `connectAndSign` method provides a streamlined approach for dapps to interact
with MetaMask.
```

File: wallet/how-to/sign-data/connect-and-sign.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+This method combines the [`eth_requestAccounts`] and [`personal_sign`] RPC methods, executing them sequentially.
+`connectAndSign` takes one parameter, the message string to be signed, and passes the message and
+the output of [`eth_requestAccounts`] directly to [`personal_sign`].
+
+This streamlined method improves dapp user experience, especially on mobile platforms, since it
+requests the user to connect to MetaMask and sign a message in a single interaction, requiring only
+one switch between the mobile dapp and MetaMask Mobile.
```
Comment (bgravenorst) [inline wallet/how-to/sign-data/connect-and-sign.md:19]:
```suggestion
This method enhances dapp user experience, especially on mobile platforms, by allowing users to
connect to MetaMask and sign a message in one interaction, requiring only one switch between the
mobile dapp and MetaMask Mobile.
```

File: wallet/how-to/sign-data/connect-and-sign.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
+---
+description: Use MetaMask SDK to connect and sign in a single interaction.
+sidebar_position: 1
+---
+
+# Connect and sign
```
Comment (bgravenorst) [inline wallet/how-to/sign-data/connect-and-sign.md:6]: Connect and sign using the SDK?

**Review by bgravenorst — APPROVED** [review]
Minor feedback

**Pull request merged**